File: TopKFrequentElements_347/solution.ts
function topKFrequent(nums: number[], k: number): number[] {
  const frequencies = new Map();

  for (const num of nums) {
    const count = frequencies.get(num)?? 0;
    frequencies.set(num, count + 1);
  }

  const sortedFrequencies = Array.from(frequencies.entries()).sort((a, b) => b[1] - a[1]);
  return sortedFrequencies.slice(0, k).map((entry) => entry[0]);
}

File: TopKFrequentElements_347/solution.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        var frequencies = new Dictionary<int, int>();

        foreach (var num in nums)
        {
            if (!frequencies.ContainsKey(num)) frequencies[num] = 0;

            frequencies[num]++;
        }

        return frequencies.OrderByDescending(pair => pair.Value)
            .Take(k)
            .Select(pair => pair.Key)
            .ToArray();
    }
}